src/metabase/search/filter.clj

```diff
@@ -77,6 +78,31 @@
 
 (defmethod where-clause* ::list [_ k v] [:in k v])
 
+(defn personal-collections-where-clause
```

Comment:
I think this guy should go through the indexing phase w/o cutting corners like that

Reply:
Yeah agreed - tag every index in a personal collection with the user id.

That's a larger change I'd prefer to make once there's better testing - this code is still share-able and being covered by the legacy tests.